### PR TITLE
style-guide: Clarify let-else further

### PR DESCRIPTION
**Pull request by joshtriplett**
Give some additional examples with multi-line patterns.

Make it clearer to go on to the next case if the conditions aren't met.

### CODE REVIEW

File: src/doc/style-guide/src/statements.md [hunk 1]
```diff
@@ -159,8 +159,8 @@ before the `else`.
 
 If the initializer expression is multi-line, the `else` keyword and opening
 brace of the block (i.e. `else {`) should be put on the same line as the end of
-the initializer expression, with a space between them, if all the following are
-true:
+the initializer expression, with a space between them, if and only if all the
+following are true:
```
Comment (calebcartwright) [inline src/doc/style-guide/src/statements.md:163]: I feel like the ambiguity comes from the fact that this all starts with the "if the initializer expression multi-line" up on line 160, as that could be read as the following text only being applicable if that also holds.

I'll think on some additional text proposals, but wanted to go ahead and share that.

**Review by calebcartwright — COMMENTED** [review]
This update and the examples give us the cover we need on the rustfmt side so feel free to r=me, though one inline thought left below on a potential element of clarity

File: src/doc/style-guide/src/statements.md (hunk 1 above)
Comment (joshtriplett) [reply src/doc/style-guide/src/statements.md:163]: That's true, but in the single-line case it doesn't matter if you try to apply these rules or not; the three conditions can't possibly hold for a single-line initializer. You can't have a single-line initializer expression that consists of nothing but closing parentheses, square brackets, and/or braces.

**Comment by joshtriplett** [comment]
@bors r=calebcartwright rollup

Going to go ahead and merge this with Caleb's r+; we can still make further text improvements as they arise.

**Pull request merged**